### PR TITLE
[#3196] cleaning up introduction docs

### LINKED ISSUE #3196: Move Concept information into introduction
**Describe the change you'd like to see**
There is currently only one overview page in the Concepts doc section: https://knative.dev/docs/concepts/overview/
This is confusing because it's really just a duplicate / information that should belong in the landing page. It doesn't really have any in-depth information explaining Knative concepts.
I'm proposing to remove this section until we can create an actual concepts section with information about different terms, since for now it's just adding additional topics to the nav for no real payoff.
I will open a PR to move the information from this landing page to the overview if there are no objections.

**Additional context**
https://knative.dev/docs/concepts/overview/

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #3196 

An attempt at cleaning up the landing page, removing some unnecessary files and the README inclusion.
LMK what you think or if any of this is incorrect - I cobbled it together from the two existing concept/intro type docs.

### CODE REVIEW

**Comment by omerbensaadon** [comment]
Just so I'm clear, when you say "Landing Page"  you mean knative.dev/docs, right?

If so, this is a great change!! 

/lgtm

**Review by omerbensaadon — APPROVED** [review]

**Comment by omerbensaadon** [comment]
PS: 1 less thing in the tree nav is a nice bonus

File: docs/_index.md
Comment (evankanderson) [inline docs/_index.md:25]: This sentence feels "wrong" to me, because the first half doesn't really imply the second half.  What about:

> Knative builds higher-level APIs atop the existing Kubernetes API; Knative resources can interact easily with other Kubernetes-native resources when needed. Cluster Administrators will find that existing Kubernetes tools can be used to manage Knative as well. Developers will find that common languages and frameworks which include Kubernetes-friendly tooling work smoothly with Knative and reduce the time spent solving common deployment issues:

**Review by evankanderson — COMMENTED** [review]
Sorry about getting this in late; I started taking a look on Friday but got pulled away by family.

Thanks for combining this; we should probably also look at breaking out the "learn more" into how-tos, reference, and background/architecture in a future pass, and linking in more of the eventing content (13/17 of the links are serving right now).

File: docs/_index.md
Comment (omerbensaadon) [reply docs/_index.md:10]: I think Knative does do both and we should highlight that wherever we can? I don't see the harm in leaving this as is.

File: docs/_index.md
Comment (omerbensaadon) [reply docs/_index.md:12]: I am confused by this comment

File: docs/_index.md
Comment (abrennan89) [reply docs/_index.md:10]: I sort of agree with both statements - a ton of Knative bits are specifically only for use by cluster admins, but I get maybe wording this differently. Maybe we need to split this and have two intro statements; one re what value it gives for admins, and then this one targeted at just devs?

File: docs/_index.md
Comment (abrennan89) [reply docs/_index.md:12]: @omerbensaadon s/ just means change, just a minor wording comment.

**Comment by abrennan89** [comment]
Thanks for the comments @evankanderson - I'll open a new PR to make the updates because Omer accidentally merged this one when he lgtm'd it.

File: docs/_index.md
Comment (omerbensaadon) [reply docs/_index.md:25]: I am assuming you are talking about Line 25 and on? 

I like where you are going with this, what about: 

> Knative builds more developer-friendly APIs atop the existing Kubernetes API so Knative resources can interact easily with other Kubernetes-native resources when needed. 

>Cluster Administrators will find that existing Kubernetes tools can be used to manage Knative as well. Developers will find that common languages and frameworks which include Kubernetes-friendly tooling work smoothly with Knative and reduce the time spent solving common deployment issues.

**Comment by omerbensaadon** [comment]
@abrennan89 😅

File: docs/_index.md
Comment (omerbensaadon) [reply docs/_index.md:25]: @abrennan89 ^^^

File: docs/_index.md
Comment (evankanderson) [reply docs/_index.md:10]: Yes, I think we need to pitch the roles separately. Additionally, Knative's main benefit for cluster administrators is that it makes it easier for developers to fill up (i.e. adopt) the clusters, and use them more effectively (sharing/oversubscription).

**Comment by evankanderson** [comment]
Since @abrennan89 is an approver, she only needs an `/lgtm` to have changes merged. If she wants to avoid auto-merge once there's an `/lgtm`, she can either `/approve cancel` to cancel your own implicit approval, or `/hold` to block submission until someone does a `/hold cancel`.

**Comment by evankanderson** [comment]
(It's generally rude to `/hold cancel` someone else's hold without checking with them -- I'll usually say `/hold` for XYZ as a way to say that anyone can cancel once XYZ is done.)

**Comment by abrennan89** [comment]
@evankanderson yeah I filled Omer in re my stuff being auto-approved, I didn't think he'd be able to LGTM it or I'd have added a hold before I asked for the review but I'll start doing that from now on

File: docs/_index.md
Comment (abrennan89) [reply docs/_index.md:10]: OK, removing the admin bit for now, but I do think we should revisit later when we get more time to work on persona-related stuff :)

File: docs/_index.md
Comment (abrennan89) [reply docs/_index.md:12]: I think this one should remain "that" because it's a restrictive clause: https://developers.google.com/style/pronouns?hl=en#relative-pronouns

File: docs/_index.md
Comment (abrennan89) [reply docs/_index.md:17]: Made some mods to this part in the new PR.

File: docs/_index.md
Comment (abrennan89) [reply docs/_index.md:25]: I'm going to do some editing myself on this part in the new PR and you guys can take a look. I'm not sold on either of these entirely because I think there's too much fluff in both that doesn't add value, but I get what you're trying to say.

**Pull request merged**
